boskos/metrics/metrics.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"k8s.io/test-infra/boskos/client"
+)
+
+type prometheusMetrics struct {
+	GceStats map[string]prometheus.Gauge
+	GkeStats map[string]prometheus.Gauge
+}
+
+var (
+	promMetrics = prometheusMetrics{
+		GceStats: map[string]prometheus.Gauge{
+			"free": prometheus.NewGauge(prometheus.GaugeOpts{
+				Name: "boskos_gce_project_free",
+				Help: "Number of free gce-project",
+			}),
+			"busy": prometheus.NewGauge(prometheus.GaugeOpts{
+				Name: "boskos_gce_project_busy",
+				Help: "Number of busy gce-project",
+			}),
+			"dirty": prometheus.NewGauge(prometheus.GaugeOpts{
+				Name: "boskos_gce_project_dirty",
+				Help: "Number of dirty gce-project",
+			}),
+			"cleaning": prometheus.NewGauge(prometheus.GaugeOpts{
+				Name: "boskos_gce_project_cleaning",
+				Help: "Number of cleaning gce-project",
+			}),
+		},
+		GkeStats: map[string]prometheus.Gauge{
+			"free": prometheus.NewGauge(prometheus.GaugeOpts{
+				Name: "boskos_gke_project_free",
+				Help: "Number of free gke-project",
+			}),
+			"busy": prometheus.NewGauge(prometheus.GaugeOpts{
+				Name: "boskos_gke_project_busy",
+				Help: "Number of busy gke-project",
+			}),
+			"dirty": prometheus.NewGauge(prometheus.GaugeOpts{
+				Name: "boskos_gke_project_dirty",
+				Help: "Number of dirty gke-project",
+			}),
+			"cleaning": prometheus.NewGauge(prometheus.GaugeOpts{
+				Name: "boskos_gke_project_cleaning",
+				Help: "Number of cleaning gke-project",
+			}),
+		},
+	}
+)
+
+func init() {
+	for _, gce := range promMetrics.GceStats {
+		prometheus.MustRegister(gce)
+	}
+
+	for _, gke := range promMetrics.GkeStats {
+		prometheus.MustRegister(gke)
+	}
+}
+
+func main() {
+	logrus.SetFormatter(&logrus.JSONFormatter{})
+	boskos := client.NewClient("Reaper", "http://boskos")
```

Comment:
I think the owner here should **not** be "Reaper", since this deployment should only ever use the `Client.Metric` function which doesn't require an owner. If this deployment somehow tried to acquire a resource it would appear to be owner by "Reaper" instead of showing that it was improperly acquired by this deployment which would be harder to debug.